src/django/api/processing.py

```diff
@@ -128,87 +128,72 @@ def geocode_facility_list_item(item):
-            # New Facility
-            if f_l_item.facility is None:
-                return get_new_facility_match_result(f_l_item.id, None, result)
-            if f_l_item.facility.created_from == f_l_item:
-                return get_new_facility_match_result(
-                    f_l_item.id, f_l_item.facility.id, result
-                )
 
-            # Automatic Match
-            return get_automatic_match_result(f_l_item.id,
-                                              f_l_item.facility.id,
-                                              automatic_matches[0].confidence,
-                                              context,
-                                              result)
-        else:
-            # New Facility
-            if f_l_item.facility_id is None:
-                return get_new_facility_match_result(f_l_item.id, None, result)
-
-            # Automatic Match
-            return get_automatic_match_result(f_l_item.id,
-                                              f_l_item.facility_id,
-                                              automatic_matches[0].confidence,
-                                              context,
-                                              result)
-
-    pending_matches = [
-        match for match in queryset_f_m
-        if match.status == match_object_type.PENDING
-    ]
-    for item in pending_matches:
-        # Potential Match
-        return get_potential_match_result(f_l_item,
-                                          item,
-                                          len(pending_matches),
-                                          context,
-                                          should_create,
-                                          result)
+def wait_for_match_processing(id, list_item_object_type, timeout=25):
+    end_match_statuses = {
+        list_item_object_type.MATCHED,
+        list_item_object_type.POTENTIAL_MATCH,
+        list_item_object_type.ERROR_MATCHING
+    }
+
+    for _ in range(timeout):
+        facility_list_item = list_item_object_type.objects.get(id=id)
+        if facility_list_item.status in end_match_statuses:
+            return facility_list_item
+        time.sleep(1)
+
+    return None
+
+
+def process_matches(f_l_item, matches, context, should_create, result):
+    automatic_matches = [m for m in matches if m.status == FacilityMatch.AUTOMATIC]
+    if len(automatic_matches) == 1:
+        return handle_new_facility_or_automatic_match(f_l_item, automatic_matches[0], context, result)
+
+    # Potential Matches
+    pending_matches = [m for m in matches if m.status == FacilityMatch.PENDING]
+    if pending_matches:
+        return handle_potential_matches(f_l_item, pending_matches, context, should_create, result)
 
     return result
 
 
+def handle_new_facility_or_automatic_match(f_l_item, match, context, result):
+    facility_id = getattr(f_l_item, 'facility_id', None) or getattr(f_l_item, 'facility', None)
```

Comment:
Instead of using if/else condition with`isinstance(f_l_item, FacilityListItem)`, used approach with `getattr`. This ensures that we always get the correct facility_id, whether the object is **FacilityListItem** or **FacilityListItemTemp**.